### ISSUE TITLE
Rule Proposal: unused logger in module

### ISSUE THREAD

**Issue by valentincalomme** [astral-sh/ruff#9465]
# unused-logger (RUF-0XX)

## What it does

Checks if a logger isn't used in the module it's instantiated.

## Why is it bad

Although you could instantiate a logger in one file and use it in another, this is not recommended. Therefore, if a logger isn't used in the module, it's instantiated; it should be considered unused and removed.

## Example

If I have a simple `module.py`

```py
"""A simple module."""
import logging

logger = logging.getLogger(__name__)


def foo() -> None:
    ...
```

The logger isn't used, which adds 2 unnecessary lines. The linter should fix the file so that it looks like this:

```py
"""A simple module."""


def foo() -> None:
    ...

```

## When not to use it

If you don't care about the additional lines.

**Comment by charliermarsh** [comment]
I thought it was somewhat common to define a logger and import it elsewhere -- is that not recommended?

**Comment by zanieb** [comment]
Is `logger` excluded from unused variable rules?

**Comment by charliermarsh** [comment]
Variables defined at the top-level of the module are always excluded from unused variable rules.

**Comment by ValentinCalomme** [comment]
> It is common, but not recommended by the official docs. They're clear that they recommend a new logger per module with __name__.

This is what I'm trying to enforce basically. It's an edge case specifically for any variable at the top of the module that is assigned the value `logging.getLogger(__name__)`.

I think it could be this specific as `ruff` already has other rules to enforce how to instantiate a logger (i.e., LOG0XX rules)

**Comment by jenstroeger** [comment]
I think in this context it would perhaps make sense to ask the user to use global sunder name for the module logger to make the intent clear:
```python
# This logger is private to the module.
_logger = logging.getLogger(__name__)
```
